<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>RemoteControlVISAControlDialog</class>
 <widget class="QDialog" name="RemoteControlVISAControlDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>821</width>
    <height>540</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>Liberation Sans</family>
    <pointsize>9</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>VISA Control</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="controlsGroup">
     <property name="title">
      <string>Controls</string>
     </property>
     <property name="flat">
      <bool>false</bool>
     </property>
     <layout class="QFormLayout" name="formLayout_3">
      <item row="1" column="0">
       <widget class="QLabel" name="nameLabel">
        <property name="text">
         <string>Name</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="name">
        <property name="toolTip">
         <string>Name for this control</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="idLabel">
        <property name="text">
         <string>ID</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="id">
        <property name="toolTip">
         <string>Unique identifier for this control</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="typeLabel">
        <property name="text">
         <string>Type</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <layout class="QHBoxLayout" name="typeLayout">
        <item>
         <widget class="QComboBox" name="type">
          <property name="minimumSize">
           <size>
            <width>100</width>
            <height>0</height>
           </size>
          </property>
          <item>
           <property name="text">
            <string>Boolean</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Integer</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Float</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>String</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>List</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Button</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="widgetType">
          <property name="toolTip">
           <string>Type of widget to display value</string>
          </property>
          <item>
           <property name="text">
            <string>Spin box</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Dial</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Slider</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="minLabel">
          <property name="text">
           <string>Min</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="min">
          <property name="toolTip">
           <string>Minimum value</string>
          </property>
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="minimum">
           <double>-9999999999999999932209486743616279764617084419440640.000000000000000</double>
          </property>
          <property name="maximum">
           <double>10000000000000000735758738477112498397576062152177456799245857901351759143802190202050679656153088.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="maxLabel">
          <property name="text">
           <string>Max</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="max">
          <property name="toolTip">
           <string>Maximum value</string>
          </property>
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="minimum">
           <double>-999999999999999983222784.000000000000000</double>
          </property>
          <property name="maximum">
           <double>100000000000000007629769841091887003294964970946560.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="scaleLabel">
          <property name="text">
           <string>Scale</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="scale">
          <property name="toolTip">
           <string>Scale factor applied to value before sending to instrument</string>
          </property>
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="maximum">
           <double>1000000000000000.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="precisionLabel">
          <property name="text">
           <string>Precision</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="precision">
          <property name="toolTip">
           <string>Precision (number of decimals)</string>
          </property>
          <property name="maximum">
           <number>323</number>
          </property>
          <property name="value">
           <number>3</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="labelLabel">
          <property name="text">
           <string>Label</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="label">
          <property name="toolTip">
           <string>Button label</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="values">
          <property name="minimumSize">
           <size>
            <width>150</width>
            <height>0</height>
           </size>
          </property>
          <property name="toolTip">
           <string>List of allowable values</string>
          </property>
          <property name="editable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="remove">
          <property name="toolTip">
           <string>Remove current entry from list</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="../../../sdrgui/resources/res.qrc">
            <normaloff>:/bin.png</normaloff>:/bin.png</iconset>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="unitsLabel">
        <property name="text">
         <string>Units</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QLineEdit" name="units">
        <property name="toolTip">
         <string>Units</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="setStateLabel">
        <property name="text">
         <string>Set state</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QTextEdit" name="setState"/>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="getStateLabel">
        <property name="text">
         <string>Get state</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QTextEdit" name="getState"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>name</tabstop>
  <tabstop>id</tabstop>
  <tabstop>type</tabstop>
  <tabstop>min</tabstop>
  <tabstop>max</tabstop>
  <tabstop>precision</tabstop>
  <tabstop>values</tabstop>
  <tabstop>units</tabstop>
  <tabstop>setState</tabstop>
  <tabstop>getState</tabstop>
 </tabstops>
 <resources>
  <include location="../../../sdrgui/resources/res.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>RemoteControlVISAControlDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>295</x>
     <y>619</y>
    </hint>
    <hint type="destinationlabel">
     <x>295</x>
     <y>319</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>RemoteControlVISAControlDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>295</x>
     <y>619</y>
    </hint>
    <hint type="destinationlabel">
     <x>295</x>
     <y>319</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
